source/common/upstream/cluster_manager_impl.h

```diff
@@ -574,10 +698,17 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
                               const ClusterConnectivityState& cluster_manager_state,
                               std::function<ConnectionPool::Instance*()> preconnect_pool);
 
+  ClusterDiscoveryCallbackHandlePtr
+  requestOnDemandClusterDiscovery(OdCdsApiSharedPtr odcds, const std::string& name,
+                                  ClusterDiscoveryCallbackSharedPtr callback,
+                                  std::chrono::milliseconds timeout);
+
+private:
   ClusterManagerFactory& factory_;
   Runtime::Loader& runtime_;
   Stats::Store& stats_;
   ThreadLocal::TypedSlot<ThreadLocalClusterManagerImpl> tls_;
+  ClusterCreationsMap pending_cluster_creations_;
```

Comment:
Can you add a comment on what this is for?

Reply:
Will do.